src/functions/artifacts.ts

```diff
@@ -0,0 +1,182 @@
+import * as artifactregistry from "../gcp/artifactregistry";
+import { FirebaseError } from "../error";
+
+/**
+ * Repository id used by Cloud Run functions for storing artifacts.
+ * See https://cloud.google.com/functions/docs/building#image_registry
+ */
+export const GCF_REPO_ID = "gcf-artifacts";
+
+/** ID used for cleanup policies created by Firebase CLI */
+export const CLEANUP_POLICY_ID = "firebase-functions-cleanup";
+
+/**
+ * Label key used to mark repositories as opted-out from cleanup policy
+ * This prevents prompts from being printed to encourage developers to
+ * set up cleanup policies.
+ */
+export const OPT_OUT_LABEL_KEY = "firebase-functions-cleanup-opted-out";
+
+const SECONDS_IN_DAY = 24 * 60 * 60;
+
+/**
+ * Construct the full path to a repository in Artifact Registry
+ *
+ * @returns The full path to the repository
+ */
+export function makeRepoPath(
+  projectId: string,
+  location: string,
+  repoName: string = GCF_REPO_ID,
+): string {
+  return `projects/${projectId}/locations/${location}/repositories/${repoName}`;
+}
+
+/**
+ * Extract an existing cleanup policy from the repository if it exists
+ *
+ * @returns The existing policy if found, undefined otherwise
+ */
+export function findExistingPolicy(
+  repository: artifactregistry.Repository,
+): artifactregistry.CleanupPolicy | undefined {
+  if (!repository.cleanupPolicies) {
```

Comment:
Super nit: this can all be `return repository?.cleanupPolicies?.[CLEANUP_POLICY_ID]`

Reply:
i like one liners.